imports/lib/discord.ts

```diff
@@ -3,9 +3,13 @@ import { DiscordAccountType } from './schemas/DiscordAccount';
 const DiscordOAuthScopes = ['identify', 'guilds.join'];
 const API_BASE = 'https://discord.com/api/v8';
 
+// If size is not a supported resolution, returns the next supported size up (or the max size)
 function getAvatarCdnUrl(da?: DiscordAccountType, size: number = 128): string | undefined {
   if (da?.avatar) {
-    return `https://cdn.discordapp.com/avatars/${da.id}/${da.avatar}.png?size=${size}`;
+    const supportedSizes = [16, 20, 32, 40, 60, 64, 80, 100, 128, 160, 256, 320, 512];
+    const maxSize = supportedSizes[supportedSizes.length - 1];
+    const requestSize = supportedSizes.find((s) => s >= size) ?? maxSize;
+    return `https://cdn.discordapp.com/avatars/${da.id}/${da.avatar}.png?size=${requestSize}`;
```

Comment:
Are there cases where we actually want to render an arbitrary-sized avatar, or does it make sense to require passing an exact size instead? Something like

```ts
export type VALID_AVATAR_SIZE = 16 | 20 | 32 | 40 | ...;

function getAvatarCdnUrl(da?: DiscordAccountType, size: VALID_AVATAR_SIZE = 128: string | undefined {
...
}
```

would make it a type error to pass in an unsupported size (and you could use `VALID_AVATAR_SIZE` elsewhere that a size was being thrown around)

Reply:
There actually is a big case for requesting an arbitrary size: hiDPI assets. For instance, in the chat, we request a 40px avatar, which makes calls for 40px, 80px, and 120px, the last of which isn't an available size and gets bumped to 128px.

I'm happy to add `VALID_AVATAR_SIZE` as you suggest. but I'd argue we still need to add a companion utility method to get the next available size. (Since `Avatar` is now the only thing that calls `getAvatarCdnUrl`, that utility method would always be invoked and it wouldn't change much, but I see the argument that this improves the discord library.) Assuming we went that way, would you prefer the utility method be a stand-alone size calculator or wrap `getAvatarCdnUrl`?

Reply:
Ah OK I hadn't considered hiDPI. I'm OK leaving this as-is given that.